### DISCUSSION TITLE
Open in terminal in mixplorer always open home of termux

### DISCUSSION THREAD

**Discussion by elswerky** [termux/termux-packages#7537] [Q&A] [answered]
Open in terminal in mixplorer always open home of termux
Whatever the path am in in mixplorer ,when I choose open in terminal and choose termux,it always opens home of termux 
I want it to open current path am in in mixplorer
I installed another terminal and found it working in it

**Comment by agnostic-apollo** [comment]
Have you granted termux storage permission? If you have and are on android 11, disable and grant it again.

**Reply by elswerky** [reply]
Yes granted permessions
And removed it and granted for some times 
It always open 
/data/data/com.termux/files/home

I installed another terminal app which is about 300 on only and not updated since 2016 but it doesn't have have this issue  and always open working directory unless I defined certain oath or command 

But termux not responding or listening to mixplorer commands

**Reply by elswerky** [reply]
Yes on  Android 11

**Comment by agnostic-apollo** [comment, marked as answer]
So I tested with `v6.57.1-API26-RC` and its passing the `top` binary as executable and even that in background mode. A user reported earlier about executable not being set in termux/termux-app#2135 and now this. I have no idea what the dev is trying to do here, should report to him. The `login` binary should be passed as executable and background mode extra should be `false`. Working directory is being correctly passed it seems. The details are at https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent, and dev is likely copy/pasting examples. Since `top` command is not being passed any arguments to end, it will run indefinitely in the background, note running tasks in termux notification.

You can check by setting log level to verbose in termux settings and running `logcat` command and then monitoring what intent data MiXplorer sends.

**Reply by elswerky** [reply]
I found that the version I use of termux is old version from olay store and that I need to update from fdroid 
Maybe this is the issue 
Will update termux from.fdroid and see may be this is the problem (maybe )

**Reply by agnostic-apollo** [reply]
You should definitely install from F-Droid now but the above issue is not related to Termux, I test `RUN_COMMAND` intent quite frequently and did couple of days ago and is working fine.

https://github.com/termux/termux-app#google-play-store-deprecated

**Reply by elswerky** [reply]
![Screenshot_20210915-074122907](https://user-images.githubusercontent.com/26687459/133385242-7ca340f8-ef65-4935-bb6c-441db6b485ae.jpg)


I got this issue when I run open in terminal.in mixplorer 

This issue is from termux
Current installed version is latest version from.fdroid.

**Reply by agnostic-apollo** [reply]
You haven't posted the actual error. And likely issue is `allow-external-apps`. Read https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent#Setup-Instructions

**Reply by elswerky** [reply]
First error was allow-external-apps 
So put
allow_external_apps=true 
In
 ~/.termux/termux.properities 
Don't know if this right or wrong 
Now.i got another error which is 
com.termux.RUN_COMMAND_PATH

Really I don't know what to do 
And I read page about it but didn't understand what to do 
And even if I fixed this ,,may be still termux can't open specefied  path in mixplorer


Even the store version I didn't face such errors in it , although it was always opening default termux path but never faced such problems

**Reply by elswerky** [reply]
![Screenshot_20210915-124918983](https://user-images.githubusercontent.com/26687459/133420376-d55cbbbb-b381-4eca-9335-dc8e91c0b12e.jpg)

**Reply by elswerky** [reply]
I added allow-external-apps=true in ~/.termux/termux.properities 

Now this error disappeared but appeared new error related to 
_**Mandatory extra missing to RunCommandService: "com.termux.RUN_COMMAND_PATH"**_

I read wiki page of this error but don't know or understand how to fix it 

If you can help 

The play store version I didn't face such errors in it

**Reply by agnostic-apollo** [reply]
That's the same error as the one in termux/termux-app#2135 and is an issue with MiXplorer, not Termux.

You won't see errors is playstore version because its quite old and doesn't have the checks/validation like the F-Droid one does. It will not work in playstore version either even if no error is shown.

Report to MiXplorer dev, I have nothing more to add here.

**Reply by elswerky** [reply]
Ok ,I know really I thank you for your support and replies and am really embarrassed to keep replying you and wasting your time 
So thank you for your support and reply 

Just a thing ,, you said you tested mixplorer version  v6.57.1-API26-RC. , I tested it still opening home directory of termux meaning command only opening termux 
Running a11 custom rom

So why it works with you while not working with me

**Reply by agnostic-apollo** [reply]
Welcome.

It is not working even for me. MiXplorer is trying to run the `top` command in the background which is used to show the processes using the most resources, its not a command that's used to start a new session/shell. The home directory opens cause MiXplorer manually starts the `TermuxActivity` with `launchIntentForPackage` in addition to sending `RUN_COMMAND` intent, which is not needed and will open two sessions, Termux will automatically do it. It's clearly sending wrong data to Termux...

```
StringBuilder sb;
try {
    String replace = str3.replace("$workdir", "\"" + str2 + "\"");
    Intent launchIntentForPackage = d01.h().getLaunchIntentForPackage(str);
    launchIntentForPackage.addCategory("android.intent.category.DEFAULT");
    launchIntentForPackage.addFlags(268435456);
    if ("com.termux".equals(str)) {
        launchIntentForPackage.setClassName(str, str + ".app.TermuxActivity");
        if (nhVar.c("com.termux.permission.RUN_COMMAND")) {
            boolean startsWith = replace.startsWith("/");
            if (!startsWith) {
                replace = "/data/data/com.termux/files/usr/bin/top";
            }
            if (startsWith) {
                str2 = e34.J(replace);
            }
            d01.f().postDelayed(new na1(str, startsWith, replace, str2), 30);
        } else {
            throw new kn2();
        }
    }
    ...
    xx1.a0(nhVar, launchIntentForPackage);
    return true;
}
```

```
 Intent intent = new Intent();
  intent.setClassName(this.i, this.i + ".app.RunCommandService");
  intent.setAction(this.i + ".RUN_COMMAND");
  if (this.O1) {
      str = this.i + ".RUN_COMMAND_PATH";
  } else {
      str = this.i + ".RUN_COMMAND_PATH";
  }
  intent.putExtra(str, this.P1);
  intent.putExtra(this.i + ".RUN_COMMAND_WORKDIR", this.Q1);
  intent.putExtra(this.i + ".RUN_COMMAND_BACKGROUND", true);
  intent.putExtra(this.i + ".RUN_COMMAND_SESSION_ACTION", "0");
  if (dz3.v()) {
      d01.b.startForegroundService(intent);
  } else {
      d01.b.startService(intent);
  }
```

**Reply by x2v0** [reply]
Do you have MiXplorer  sources?

**Reply by agnostic-apollo** [reply]
Everybody has them, APKs can be decompiled to get java code, which may be obfuscated.

https://github.com/skylot/jadx

**Comment by elswerky** [comment]
![Uploading Screenshot_20210915-124918983.jpg…]()
